### DISCUSSION TITLE
Does Redisson lock work for master replication nodes?

### DISCUSSION THREAD

**Discussion by sarmuru2** [Netflix/conductor#3859] [General]
We are running multiple pods of conductor statelessly where there is a possibility of invoking the WorkflowExecutor.decide() method for the same workflow from multiple threads or even from multiple pods as well. Because of this there are cases where tasks from the same workflow are scheduled twice.  To avoid this, we are trying to use RedisLock. But https://github.com/Netflix/conductor/blob/main/redis-lock/src/main/java/com/netflix/conductor/redislock/config/RedisLockConfiguration.java does not have options for Replicated redis server. We are having 3 redis nodes with master to master replication and no slave nodes. As per this link,  https://dev.to/lazypro/explain-redlock-in-depth-31jj, redisson redlock will not work for the cases like our deployment where we have master to master replication of 3 nodes. with no slave nodes. 

Please let us know to use RedisLock in the case of master to master replication of redis nodes.

**Comment by charybr** [comment]
Hi Sarvana,

For distributed locking to work Redis nodes participating in the majority consensus should be independent.  In the case you are describing they are replicas, so we cannot use RedisLock. 

Other options you can consider are:
1. Using separate set of redis nodes (which are not replicas) dedicated for locking.
2. Using zookeeper locking.
3. Using dyno queue localOnly strategy.

Option-3 above is fairly reasonable with a drawback as explained in the link below.
https://conductor.netflix.com/devguide/architecture/technicaldetails.html
